client/app/pages/dashboards/public-dashboard-page.html

```diff
@@ -15,7 +15,11 @@
     <filters ng-if="$ctrl.dashboard.dashboard_filters_enabled"></filters>
   </div>
 
-    <div ng-repeat="row in $ctrl.dashboard.widgets" class="row">
-      <dashboard-widget ng-repeat="widget in row" widget="widget" dashboard="$ctrl.dashboard" public="true"></dashboard-widget>
+
+  <div gridster="$ctrl.dashboardGridOptions">
+    <div ng-repeat="widget in $ctrl.dashboard.widgets">
+      <dashboard-widget widget="widget" gridster-item="widget.options.position"
```

Comment:
Don't we need the auto-height directive here too?

It feels like we should encapsulate the whole dashboard grid logic in its own component that will be reused here and in the regular dashboard.

Reply:
Got it, will take a look.

Reply:
I think, we should add some issues labeled as `refactoring`, and this one should be among them (as well as moving to LESS) - otherwise this PR tends to grow too large.

Reply:
👍 
It's the "Tech Debt" label. You're welcome to add those.